lib/ecto/adapters/sql.ex

```diff
@@ -249,6 +245,31 @@ defmodule Ecto.Adapters.SQL do
     end
   end
 
+  @doc """
+  TODO: adapter's specific details should be documented here?
+
+  ## Examples
+
+      iex> Ecto.Adapters.SQL.explain(:all, repo, Post)
+      "Seq Scan on posts p0  (cost=0.00..12.12 rows=1 width=443)"
+
+  """
+  @spec explain(Ecto.Repo.t, :all | :update_all | :delete_all, Ecto.Queryable.t, Keyword.t) :: String.t
+  def explain(repo, operation, queryable, opts \\ []) do
+    {prepared, params} = to_sql(operation, repo, queryable)
+    %{sql: sql} = adapter_meta = Ecto.Adapter.lookup_meta(repo)
+
+    {:ok, explain} =
+      transaction(adapter_meta, [], fn ->
+        query = sql.explain_query(prepared, opts)
+        {:ok, %{rows: rows}} = sql_call(adapter_meta, :query, [query], params, [])
+        rows
+      end)
+
+    # TODO: support more formats
+    Enum.map_join(explain, "\n", & &1)
```

Comment:
Calling `IO.puts` on that string will pretty print as returned by the database, but is that enough for the first version? I'm not sure 🤔

Reply:
I think so :+1:.